gordo_components/workflow/config_elements/machine.py

```diff
@@ -71,6 +71,11 @@ def from_config(  # type: ignore
         model = config.get("model") or config_globals.get("model")
 
         local_runtime = config.get("runtime", dict())
+        if "server" in local_runtime:
+            raise ValueError(
+                "Runtime settings for the model-server must be set as globals, not at machine level."
+            )
+
```

Comment:
This _seems_ like it should rather be in the descriptor (`ValidMachineRuntime`) as that's where we're doing all validation of what constitutes a 'valid' runtime for a machine. 

Also, with that said, it seems to me that maybe this shouldn't raise an error, but a deprecation warning if anything? I can stick `foo` into the runtime as well, and it wouldn't care, right? Since it looks at specific keys not the keys actually contained. But maybe it does makes sense; making it painfully obvious for people upgrading versions. But even so, do we think anyone _actually_ sets the server runtime?

Reply:
>This seems like it should rather be in the descriptor (ValidMachineRuntime) as that's where we're doing all validation of what constitutes a 'valid' runtime for a machine.

Make sense :+1: 

On :zap: Warning:zap:   VS :boom: Exception:boom:, @epa095 preferred hard compliance here, and after thinking about it a bit I think I'm on-board with that. If said warning is not noticed, it could make for some confusing debugging.

Reply:
Tried out doing things as you suggested for the first one, but then the machine validator fails on the default server resources that get injected to the machine.

Reply:
Then I suggest we fix that. Because doing a validation here, is not the correct place when we have all other verification of attributes in the validators.

The possibilities I see here is that `DEFAULT_CONFIG_GLOBALS` should maybe be split into what is a _valid_ `Machine` global, and a valid `global` as it were is a superset of a valid Machine global. Because it appears these are now two different things. And then update the tests to use the appropriate `global` default config.

Ooorrrr. just ignore the key, like we would ignore _any other_ key which we don't care about and this whole problem goes away.